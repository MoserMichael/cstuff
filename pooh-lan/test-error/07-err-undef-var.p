

sub foo()

   println( ~msg undefined_local )

   println( ~msg no_such_global )


   return sub ()
        println( ~msg outer.no_capture )
     end
end
