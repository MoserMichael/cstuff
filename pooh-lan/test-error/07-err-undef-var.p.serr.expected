
test-error/07-err-undef-var.p(5): Error: local variable undefined_local is used without having been assigned a value
      5 |:....+....:....+....:....+....:....+....:....+....:....+....:....+....:....+....
        |   println( ~msg undefined_local )
        |.................^^^^^^^^^^^^^^^

test-error/07-err-undef-var.p(7): Error: local variable no_such_global is used without having been assigned a value
      7 |:....+....:....+....:....+....:....+....:....+....:....+....:....+....:....+....
        |   println( ~msg no_such_global )
        |.................^^^^^^^^^^^^^^

test-error/07-err-undef-var.p(11): Error: Closure variable no_capture is used without having been assigned a value
     11 |:....+....:....+....:....+....:....+....:....+....:....+....:....+....:....+....
        |        println( ~msg outer.no_capture )
        |......................^^^^^^^^^^^^^^^^
Type checking failed
